torch/csrc/autograd/python_variable_indexing.cpp

```diff
@@ -468,12 +469,16 @@ int THPVariable_setitem(PyObject* self, PyObject* index, PyObject* py_value) {
   bool is_tracing = torch::jit::tracer::isTracing();
 
   // handle simple types: integers, slices
-  if (THPUtils_checkLong(index)) {
+  if (THPUtils_checkLong(index) || torch::is_symint(index)) {
     if (is_tracing && THPVariable_Check(index)) {
       recordSelectTrace(THPVariable_Unpack(index));
     }
-    dispatch_set_item(
-        self_, {at::indexing::TensorIndex(THPUtils_unpackLong(index))}, value);
+    auto symint = torch::is_symint(index)
+        ? SymInt(static_cast<SymNode>(
```

Comment:
There no simpler way to extract the SymInt from the PyObject ??

Reply:
Yeah do we not have a working cast method implementation that does? @ezyang ?

Reply:
py::cast should work E.g. `return py::cast<c10::SymInt>(index);`